src/prefabs/pageWithEditDataTable.js

```diff
@@ -0,0 +1,11725 @@
+(() => ({
+  name: 'Page With Edit Data Table',
+  icon: 'DataTable',
+  type: 'page',
+  description: 'This page contains a data table with edit dialog',
+  category: 'DATA',
+  beforeCreate: ({
+    helpers: { useModelQuery, camelToSnakeCase },
+    components: {
+      Content,
+      Header,
+      Field,
+      Footer,
+      ModelSelector,
+      PropertiesSelector,
+      CheckBox,
+      Text,
+      Box,
+    },
+    prefab,
+    save,
+    close,
+  }) => {
+    const [modelId, setModelId] = React.useState('');
+    const [properties, setProperties] = React.useState([]);
+    const [modelValidation, setModelValidation] = React.useState(false);
+    const [propertiesValidation, setPropertiesValidation] = React.useState(
+      false,
+    );
+    const [
+      editPropertiesValidation,
+      setEditPropertiesValidation,
+    ] = React.useState(false);
+    const [editProperties, setEditProperties] = React.useState([]);
+    const [inheritProperties, setInheritProperties] = React.useState(true);
+
+    const { data } = useModelQuery({
+      variables: { id: modelId },
+      skip: !modelId,
+    });
+
+    const reduceStructure = (refValue, structure) =>
```

Comment:
This is just a suggestion, but since this is used to find a certain prefab in the structure, maybe it's better to call it something else than `reduceStructure`. I noticed it in the other PR as well. https://github.com/bettyblocks/material-ui-component-set/pull/1045. It's not wrong, but doesn't give any info either.

Reply:
Renamed `reduceStructure` to `getDescendantByRef`